tests/fast-integration/monitoring/grafana.js

```
@@ -15,75 +14,74 @@ const {
-  // Checking that authentication works and redirects to grafana URL
-  res = await assertGrafanaRedirect('https://grafana.');
-  assert.isTrue(res, 'Grafana redirect to grafana landing page does not work!');
 
-  res = await resetProxy();
-  assert.isTrue(res, 'Grafana Authproxy is not reset successfully!  ');
+  info('Checking grafana redirect to grafana URL');
+  res = await checkGrafanaRedirect('https://grafana.', 200);
+  assert.isTrue(res, 'Grafana redirect to grafana landing page does not work!');
 }
 
-async function checkGrafanaRedirectsInKyma1() {
-  const res = await assertGrafanaRedirect('https://dex.');
+async function assertGrafanaRedirectsInKyma1() {
+  info('Checking grafana redirect for dex');
+  const res = await checkGrafanaRedirect('https://dex.', 200);
   assert.isTrue(res, 'Grafana redirect to dex does not work!');
 }
 
-async function assertGrafanaRedirect(redirectURL) {
-  const vs = await getVirtualService('kyma-system', 'monitoring-grafana');
-  const hosts = vs.spec.hosts[0];
-  let ignoreSSL = false;
-  if (hosts.includes('local.kyma.dev')) {
-    ignoreSSL = true;
-  }
-  const url = 'https://' + hosts;
-  if (redirectURL.includes('https://dex.')) {
-    console.log('Checking redirect for dex');
-    return await retryUrl(url, redirectURL, ignoreSSL, 200);
-  }
-
-  if (redirectURL.includes('https://kyma-project.io/docs')) {
-    console.log('Checking redirect for kyma docs');
-    return await retryUrl(url, redirectURL, ignoreSSL, 403);
+async function setGrafanaProxy() {
+  if (getEnvOrDefault('KYMA_MAJOR_VERSION', '2') === '2') {
+    await manageSecret('create');
+    await restartProxyPod();
+    await updateProxyDeployment('--reverse-proxy=true', '--trusted-ip=0.0.0.0/0');
+    info('Checking grafana redirect to grafana URL');
+    const res = await checkGrafanaRedirect('https://grafana.', 200);
+    assert.isTrue(res, 'Grafana redirect to grafana landing page does not work!');
```

I don't think we should assert inside helper functions? Maybe throw an exception?

---

We also assert in other helper functions

---

Still think it hinders test glanceability, but let's keep it for now